[opencolorio:x64-windows] build failure

**Host Environment**
- OS: Windows
- Compiler: Visual Studio 17.0.4

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install --triplet=x64-windows opencolorio[tools]`

All dependencies of opencolorio were installed successfully, but compilation of opencolorio failed at linking stage.
Build ok if tools are not requested.

**Failure logs** 

-Console output

```
Starting package 6/6: opencolorio:x64-windows
Building package opencolorio[core,tools]:x64-windows...
-- Using cached AcademySoftwareFoundation-OpenColorIO-v2.1.1.tar.gz.
-- Cleaning sources at C:/DevLocal/vcpkg/vcpkg/buildtrees/opencolorio/src/v2.1.1-2eff5bc6b9.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/DevLocal/vcpkg/vcpkg/downloads/AcademySoftwareFoundation-OpenColorIO-v2.1.1.tar.gz
-- Applying patch fix-dependency.patch
-- Using source at C:/DevLocal/vcpkg/vcpkg/buildtrees/opencolorio/src/v2.1.1-2eff5bc6b9.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j9
    Working Directory: C:/DevLocal/vcpkg/vcpkg/buildtrees/opencolorio/x64-windows-dbg
    See logs for more information:
      C:\DevLocal\vcpkg\vcpkg\buildtrees\opencolorio\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/opencolorio/portfile.cmake:39 (vcpkg_cmake_install)
  scripts/ports.cmake:145 (include)

Error: Building package opencolorio:x64-windows failed with: BUILD_FAILED
```

-Additional failure logs in file "vcpkg\buildtrees\opencolorio\install-x64-windows-dbg-out.log"

```
...
FAILED: src/apps/ociocheck/ociocheck.exe 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\apps\ociocheck\CMakeFiles\ociocheck.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe  src\apps\ociocheck\CMakeFiles\ociocheck.dir\main.cpp.obj  /out:src\apps\ociocheck\ociocheck.exe /implib:src\apps\ociocheck\ociocheck.lib /pdb:src\apps\ociocheck\ociocheck.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL /subsystem:console  src\apputils\apputils.lib  src\OpenColorIO\OpenColorIO.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK Pass 1: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe src\apps\ociocheck\CMakeFiles\ociocheck.dir\main.cpp.obj /out:src\apps\ociocheck\ociocheck.exe /implib:src\apps\ociocheck\ociocheck.lib /pdb:src\apps\ociocheck\ociocheck.pdb /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL /subsystem:console src\apputils\apputils.lib src\OpenColorIO\OpenColorIO.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\apps\ociocheck\CMakeFiles\ociocheck.dir/intermediate.manifest src\apps\ociocheck\CMakeFiles\ociocheck.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library src\apps\ociocheck\ociocheck.lib and object src\apps\ociocheck\ociocheck.exp
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_DEFAULT" (?ROLE_DEFAULT@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_REFERENCE" (?ROLE_REFERENCE@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_DATA" (?ROLE_DATA@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_COLOR_PICKING" (?ROLE_COLOR_PICKING@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_SCENE_LINEAR" (?ROLE_SCENE_LINEAR@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_COMPOSITING_LOG" (?ROLE_COMPOSITING_LOG@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_COLOR_TIMING" (?ROLE_COLOR_TIMING@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_TEXTURE_PAINT" (?ROLE_TEXTURE_PAINT@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_MATTE_PAINT" (?ROLE_MATTE_PAINT@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_RENDERING" (?ROLE_RENDERING@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_INTERCHANGE_SCENE" (?ROLE_INTERCHANGE_SCENE@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::ROLE_INTERCHANGE_DISPLAY" (?ROLE_INTERCHANGE_DISPLAY@OpenColorIO_v2_1@@3PEBDEB)
src\apps\ociocheck\ociocheck.exe : fatal error LNK1120: 12 unresolved externals
[181/185] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe  src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir\main.cpp.obj  /out:src\apps\ociomakeclf\ociomakeclf.exe /implib:src\apps\ociomakeclf\ociomakeclf.lib /pdb:src\apps\ociomakeclf\ociomakeclf.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL /subsystem:console  src\apputils\apputils.lib  src\OpenColorIO\OpenColorIO.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
FAILED: src/apps/ociomakeclf/ociomakeclf.exe 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe  src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir\main.cpp.obj  /out:src\apps\ociomakeclf\ociomakeclf.exe /implib:src\apps\ociomakeclf\ociomakeclf.lib /pdb:src\apps\ociomakeclf\ociomakeclf.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL /subsystem:console  src\apputils\apputils.lib  src\OpenColorIO\OpenColorIO.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK Pass 1: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir\main.cpp.obj /out:src\apps\ociomakeclf\ociomakeclf.exe /implib:src\apps\ociomakeclf\ociomakeclf.lib /pdb:src\apps\ociomakeclf\ociomakeclf.pdb /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL /subsystem:console src\apputils\apputils.lib src\OpenColorIO\OpenColorIO.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir/intermediate.manifest src\apps\ociomakeclf\CMakeFiles\ociomakeclf.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library src\apps\ociomakeclf\ociomakeclf.lib and object src\apps\ociomakeclf\ociomakeclf.exp
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::METADATA_DESCRIPTION" (?METADATA_DESCRIPTION@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::METADATA_INPUT_DESCRIPTOR" (?METADATA_INPUT_DESCRIPTOR@OpenColorIO_v2_1@@3PEBDEB)
main.cpp.obj : error LNK2001: unresolved external symbol "char const * const OpenColorIO_v2_1::METADATA_OUTPUT_DESCRIPTOR" (?METADATA_OUTPUT_DESCRIPTOR@OpenColorIO_v2_1@@3PEBDEB)
src\apps\ociomakeclf\ociomakeclf.exe : fatal error LNK1120: 3 unresolved externals
[182/185] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\apps\ociobakelut\CMakeFiles\ociobakelut.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe  src\apps\ociobakelut\CMakeFiles\ociobakelut.dir\main.cpp.obj src\apps\ociobakelut\CMakeFiles\ociobakelut.dir\ocioicc.cpp.obj  /out:src\apps\ociobakelut\ociobakelut.exe /implib:src\apps\ociobakelut\ociobakelut.lib /pdb:src\apps\ociobakelut\ociobakelut.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL /subsystem:console  src\apputils\apputils.lib  C:\DevLocal\vcpkg\vcpkg\installed\x64-windows\debug\lib\lcms2.lib  src\OpenColorIO\OpenColorIO.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
[183/185] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\apps\ociowrite\CMakeFiles\ociowrite.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe  src\apps\ociowrite\CMakeFiles\ociowrite.dir\main.cpp.obj  /out:src\apps\ociowrite\ociowrite.exe /implib:src\apps\ociowrite\ociowrite.lib /pdb:src\apps\ociowrite\ociowrite.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL /subsystem:console  src\apputils\apputils.lib  src\OpenColorIO\OpenColorIO.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
[184/185] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\apps\ociochecklut\CMakeFiles\ociochecklut.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe  src\apps\ociochecklut\CMakeFiles\ociochecklut.dir\main.cpp.obj  /out:src\apps\ociochecklut\ociochecklut.exe /implib:src\apps\ociochecklut\ociochecklut.lib /pdb:src\apps\ociochecklut\ociochecklut.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL /subsystem:console  src\libutils\oglapphelpers\OpenColorIOoglapphelpers.lib  src\apputils\apputils.lib  src\OpenColorIO\OpenColorIO.lib  opengl32.lib  glu32.lib  C:\DevLocal\vcpkg\vcpkg\installed\x64-windows\debug\lib\glew32d.lib  C:\DevLocal\vcpkg\vcpkg\installed\x64-windows\debug\lib\freeglutd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
ninja: build stopped: subcommand failed.
```

--- a/ports/opencolorio/fix-buildTools.patch
+++ b/ports/opencolorio/fix-buildTools.patch
@@ -1,0 +1,13 @@
+diff --git a/src/OpenColorIO/CMakeLists.txt b/src/OpenColorIO/CMakeLists.txt
+index b992bbc..804d3c5 100644
+--- a/src/OpenColorIO/CMakeLists.txt
++++ b/src/OpenColorIO/CMakeLists.txt
+@@ -248,7 +248,7 @@ if(NOT BUILD_SHARED_LIBS)
+ else()
+     # Only exports selected symbols.
+     target_compile_definitions(OpenColorIO
+-        PUBLIC
++        PRIVATE
+             OpenColorIO_EXPORTS
+     )
+ endif()

--- a/ports/opencolorio/portfile.cmake
+++ b/ports/opencolorio/portfile.cmake
@@ -6,6 +6,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-dependency.patch
+        fix-buildTools.patch
 )
 
 file(REMOVE "${SOURCE_PATH}/share/cmake/modules/Findexpat.cmake")

--- a/ports/opencolorio/vcpkg.json
+++ b/ports/opencolorio/vcpkg.json
@@ -1,9 +1,10 @@
 {
   "name": "opencolorio",
   "version-semver": "2.1.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "OpenColorIO (OCIO) is a complete color management solution geared towards motion picture production with an emphasis on visual effects and computer animation. OCIO provides a straightforward and consistent user experience across all supporting applications while allowing for sophisticated back-end configuration options suitable for high-end production usage. OCIO is compatible with the Academy Color Encoding Specification (ACES) and is LUT-format agnostic, supporting many popular formats.",
   "homepage": "https://opencolorio.org/",
+  "license": "BSD-3-Clause License",
   "supports": "!uwp",
   "dependencies": [
     "expat",

--- a/ports/opencolorio/vcpkg.json
+++ b/ports/opencolorio/vcpkg.json
@@ -4,7 +4,7 @@
   "port-version": 2,
   "description": "OpenColorIO (OCIO) is a complete color management solution geared towards motion picture production with an emphasis on visual effects and computer animation. OCIO provides a straightforward and consistent user experience across all supporting applications while allowing for sophisticated back-end configuration options suitable for high-end production usage. OCIO is compatible with the Academy Color Encoding Specification (ACES) and is LUT-format agnostic, supporting many popular formats.",
   "homepage": "https://opencolorio.org/",
-  "license": "BSD-3-Clause License",
+  "license": "BSD-3-Clause",
   "supports": "!uwp",
   "dependencies": [
     "expat",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5002,7 +5002,7 @@
     },
     "opencolorio": {
       "baseline": "2.1.1",
-      "port-version": 1
+      "port-version": 2
     },
     "opencolorio-tools": {
       "baseline": "1.1.1",

--- a/versions/o-/opencolorio.json
+++ b/versions/o-/opencolorio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "80e8a46f8c1c2cd748834cd225edbe127a489d5a",
+      "version-semver": "2.1.1",
+      "port-version": 2
+    },
+    {
       "git-tree": "c68079d3d055b028af29997748784c7fa8457f7c",
       "version-semver": "2.1.1",
       "port-version": 1

